pal_event_server/lib/src/allowlist.dart

```diff
@@ -14,129 +14,174 @@
+            appContentRule.matches(event.responses[appContentKey])) {
+          allowAppContents = true;
+          break;
+        }
+      }
+      if (!allowAppContents) {
+        hashAllAppContentFields(event);
+      }
+    }
+    return event;
+  }
 
-        responsesAllowlisted
-            .add(<String, dynamic>{'name': 'apps_used', 'answer': data[0]});
-        responsesAllowlisted
-            .add(<String, dynamic>{'name': 'apps_used_raw', 'answer': data[1]});
-        responsesAllowlisted
-            .add(<String, dynamic>{'name': 'app_content', 'answer': data[2]});
-        responsesAllowlisted
-            .add(<String, dynamic>{'name': 'url', 'answer': data[3]});
+  void hashAllAppLoggerFields(Event event) {
+    _logger.info("hashing all app fields");
+    var responses = event.responses;
+    var appsUsedValue = responses[appsUsedKey];
+    if (appsUsedValue == null) {
+      appsUsedValue = "";
+    }
+    var appsUsedValueHash = hash(appsUsedValue);
+    responses[appsUsedKey] = appsUsedValueHash;
+    var appContentValue = responses[appContentKey];
+    if (appContentValue == null) {
+      appContentValue = "";
+    }
+    var appContentValueHash = hash(appContentValue);
+    responses[appContentKey] = appContentValueHash;
+    responses[appsUsedRawKey] = appsUsedValueHash + ":" + appContentValueHash;
+  }
 
-        event['responses'] = responsesAllowlisted;
-        results.add(event);
+  final chatRegex = RegExp(r'\bchat\b', caseSensitive: false);
+  final meetRegex = RegExp(r'\bmeet\b', caseSensitive: false);
+  final mailRegex = RegExp(r'\bmail\b', caseSensitive: false);
+  final calendarRegex = RegExp(r'\bcalendar\b', caseSensitive: false);
+  final googleDocsRegex = RegExp(r'\bGoogle Docs\b', caseSensitive: false);
+
+  void wipeDetailsOnEvent(Event event) {
+    if (event.responses.containsKey(appContentKey)) {
+      var app_content = event.responses[appContentKey];
+      if (chatRegex.hasMatch(app_content)) {
+        event.responses[appContentKey] = 'Chat';
+      } else if (meetRegex.hasMatch(app_content)) {
+        event.responses[appContentKey] = 'Meet';
+      } else if (mailRegex.hasMatch(app_content)) {
+        event.responses[appContentKey] = 'Mail';
+      } else if (calendarRegex.hasMatch(app_content)) {
+        event.responses[appContentKey] = 'Calendar';
+      } else if (googleDocsRegex.hasMatch(app_content)) {
+        event.responses[appContentKey] = 'Google Docs';
+      }
+    
+      var apps_used = '';
+      if (event.responses.containsKey(appsUsedKey)) {
+        apps_used = event.responses[appsUsedKey];
       }
+      event.responses[appsUsedRawKey] =
+          apps_used + ':' + event.responses[appContentKey];
+    }
+  }
+
+  String hash(String value) {
+    if (value == null) {
+      value = "";
     }
+    return sha1.convert(utf8.encode(value)).toString();
+  }
 
-    return results;
+  void hashAllAppContentFields(Event event) {
```

Comment:
I looked at it and there is not much shared because the appsUsedRaw field is unhashed in this function but it is hashed in the other. So, it can be refactored out but it isn't much of a win for deduplication. If you feel strongly I can do it.